animation-glib/stepper/linear.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * animation-glib/stepper/linear.cpp
+ *
+ * Copyright 2018 Endless Mobile, Inc.
+ *
+ * libanimation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * libanimation is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with eos-companion-app-service.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * GObject wrapper for linear stepper.
+ */
+
+#include <animation/stepper/linear.h>
+#include <animation/stepper/stepper.h>
+
+#include <animation-glib/constructor-helpers.h>
+#include <animation-glib/stepper/linear.h>
+#include <animation-glib/stepper/stepper-holder.h>
+
+namespace as = animation::stepper;
+
+struct _AnimationLinearStepper
+{
+  GObject parent_instance;
+};
+
+typedef struct _AnimationLinearStepperPrivate
+{
+} AnimationLinearStepperPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (AnimationLinearStepper,
+                            animation_linear_stepper,
+                            ANIMATION_TYPE_STEPPER_HOLDER)
+
+enum {
+  PROP_0,
+  PROP_LENGTH,
+  NPROPS
+};
+
+static GParamSpec *animation_linear_stepper_properties[NPROPS];
+
+static void
+animation_linear_stepper_set_property (GObject      *object,
+                                       guint         prop_id,
+                                       const GValue *value,
+                                       GParamSpec   *pspec)
+{
+  switch (prop_id)
+    {
+    case PROP_LENGTH:
+      break;
```

Comment:
Isn't there something missing here?

Reply:
Not really. the "length" property is only writable on construction.  Though perhaps I should jettison the idea of using a closure here and just make the stepper an actual object with a writable length property :/